docs/use_cases/real_time_apps/part1.md

```diff
@@ -0,0 +1,502 @@
+    cell_ids.sort_unstable();
+
+    Ok(Some(cell_ids))
+}
+
+// Transforms spreadsheet coordinates to cell id's
+// `A0 -> 0, B0 -> 1, C0 -> 2 ...`
+fn cell_references_to_ids(crf: &str) -> Option<i64> {
+    let mut col = 0;
+    let mut row = 0;
+    for c in crf.chars() {
+        if c.is_ascii_alphabetic() {
+            col = col * 26 + (c.to_ascii_uppercase() as i64 - 'A' as i64);
+        } else if c.is_ascii_digit() {
+            row = row * 10 + (c as i64 - '0' as i64);
+        } else {
+            return None;
+        }
+    }
+    Some(col + row * 26)
+}
+```
+
+We also add the following udf.toml to pull in crates from the Rust ecosystem:
+
+```toml
+xlformula_engine =  { git = "https://github.com/gz/XLFormula-Engine.git", rev = "3b39201" }
+log = "0.4"
+chrono = "0.4"
+```
+
+We use the [xlformula_engine](https://docs.rs/xlformula_engine) crate to parse formulas, like ``=A0+B0`, into a syntax tree:
+
+```ascii
+         Operation
+       /          \
+Reference(A0)   Reference(B0)
+```
+
+Our `mentions` function walks this tree and collects every Reference into a references array. A helper function, `cell_references_to_ids`, converts spreadsheet coordinates (e.g., `A0`, `B0`) to numeric cell IDs.
+
+Once the UDF is in place, we can run the pipeline, activate the changestream for `latest_cells`, and observe output like this (some columns omitted):
+
+| id  | raw_value | mentions       | ts                  |
+| --- |:--------- |:-------------- |:------------------- |
+| 0   | 41        | [ null ]       | 1970-01-01 00:00:01 |
+| 1   | 1         | [ null ]       | 1970-01-01 00:00:02 |
+| 2   | =B0       | [ null, 1 ]    | 1970-01-01 00:00:03 |
+| 3   | =A0+C0    | [ null, 0, 2 ] | 1970-01-01 00:00:04 |
+
+Notice null appears in every mentions array. This ensures the array isn’t empty for cells that don’t reference anything, preventing them from being excluded in future joins.
+
+:::info
+
+To see the latest_cells view in action, you can insert a newer value for cell 0:
+
+```sql
+insert into spreadsheet_data values (0, 0, '2025-01-01T00:00:00', '0', 0)
+```
+
+You should see two changes being emitted in the `latest_cell`
+view, the removal of the cell with value `42` and a new cell with value `0`.
+
+This setup also allows us to easily undo any changes made (by just deleting the most recent entry for a given cell).
+:::
+
+### Find the values of referenced cells
+
+Now that we know which cells are referenced by any given cell, the next step is to retrieve the values of those referenced cells. Having both the referenced IDs and their values makes it straightforward to compute formulas later.
+
+We accomplish this by:
+
+* Unnesting the `mentions` array from `latest_cells`.
+* Cross-joining the unnested `mentions` with `latest_cells`.
+* Rejoining with `spreadsheet_data` to look up each referenced cell’s `raw_value`.
+
+```sql
+create view latest_cells_with_mentions as
```

Comment:
For consistency with the rest of the docs, let's capitalize sql keywords (CREATE VIEW latest_cells_with_mentions AS...)